mcux/hwmv2.cmake

```diff
@@ -0,0 +1,41 @@
+# When this module completes, the following variables will be defined:
+# MCUX_DEVICE: SOC name, suffixed by core name when using a dual core part.
+#              Example: MIMXRT595S_cm33, or LPC55S36
+# MCUX_CPU: "CPU"+ SOC part number, followed by core name when using a dual core part.
+#              Example: CPU_MIMXRT595SFAWC_cm33, or  CPU_LPC55S36JBD100
+# MCU_DEVICE_PATH: SOC name without core suffix. Must match the name of the
+#              folder in MCUX HAL. IE MIMXRT595S, or LPC55S36
+
+
+# For most SOCs, the below settings will suffice
+string(TOUPPER ${CONFIG_SOC} MCUX_DEVICE_PATH)
+string(TOUPPER ${CONFIG_SOC} MCUX_DEVICE)
+set(MCUX_CPU CPU_${CONFIG_SOC_PART_NUMBER})
+
+# Handle multi core parts, based on the CONFIG_SOC_XXX settings in Zephyr
+if(CONFIG_SOC_MIMXRT1176_CM7)
+  set(MCUX_CORE_SUFFIX _cm7)
+elseif(CONFIG_SOC_MIMXRT1176_CM4)
+  set(MCUX_CORE_SUFFIX _cm4)
+elseif(CONFIG_SOC_MIMXRT1166_CM7)
+  set(MCUX_CORE_SUFFIX _cm7)
+elseif(CONFIG_SOC_MIMXRT1166_CM4)
+  set(MCUX_CORE_SUFFIX _cm4)
+elseif(CONFIG_SOC_LPC55S69_CPU0)
+  set(MCUX_CORE_SUFFIX _cm33_core0)
+elseif(CONFIG_SOC_LPC55S69_CPU1)
+  set(MCUX_CORE_SUFFIX _cm33_core1)
+elseif(CONFIG_SOC_LPC54114_M4)
+  set(MCUX_CORE_SUFFIX _cm4)
+elseif(CONFIG_SOC_LPC54114_M0)
+  set(MCUX_CORE_SUFFIX _cm0plus)
+elseif(CONFIG_SOC_MIMXRT685S_CM33)
+  set(MCUX_CORE_SUFFIX _cm33)
+elseif(CONFIG_SOC_MIMXRT595S_CM33)
+  set(MCUX_CORE_SUFFIX _cm33)
+endif()
```

Comment:
For hardware model v2, we will use CONFIG_SOC to figure the CPU on SoC with multiple CPU clusters, for example on imx8mm, use "CONFIG_SOC="mimx8mm6_ca53"" for Cortex A53 Core, and "CONFIG_SOC="mimx8mm6_cm4""  for M4 Core, so we need to add more device list at here, I am thinking whether we could use a script to do it, so that we will not need to update at here if we want to add a new device.

Reply:
I have wondered about this as well. The core suffix is very much an MCUX specific value, so we can't parse it from `CONFIG_SOC` or something similar. However, we could add a Kconfig such as `CONFIG_MCUX_CORE` in Zephyr, and use that here. Do you think that would make sense?

Reply:
Currently we have "CONFIG_SOC="mimx8mm6_ca53"" and" CONFIG_SOC_PART_NUMBER="MIMX8MM6DVTLZ_ca53", but still have no any configuration item to specify the cpu cluster which is listed in soc.yml, I suggest to add a more common item, just "CONFIG_SOC_CPU",  so that we can align all platforms, what do you think?

Reply:
We could certainly try to introduce a more common item- the reason I do not want to use a common identifier here is that the strings MCUX SDK uses to identify cores are slightly arbitrary- of course they do describe the core present on the system, but I am sure someone could argue the best string to describe a "Cortex M33" is "M33" rather than "CM33", which is what the HAL uses. A good example of this is the M0+ core on the LPC54114, which is called "cm0plus" by the HAL.

I've updated this PR to use an MCUX specific Kconfig, please let me know your thoughts